stripe/src/main/java/com/stripe/android/paymentsheet/ui/BaseSheetActivity.kt

```diff
@@ -98,6 +99,17 @@ internal abstract class BaseSheetActivity<ResultType> : AppCompatActivity() {
         // Make `bottomSheet` clickable to prevent clicks on the bottom sheet from triggering
         // `rootView`'s click listener
         bottomSheet.isClickable = true
+
+        viewModel.apiException.observe(this) { throwable ->
+            when (throwable) {
+                is APIConnectionException -> {
+                    viewModel.onApiError(getString(R.string.stripe_failure_connection_error))
+                }
+                else -> {
+                    viewModel.onApiError(throwable.localizedMessage)
+                }
+            }
+        }
```

Comment:
Wouldn't it be better to do this on the ViewModel directly?

Reply:
Hm, I wasn't sure how to get the string from the viewmodel

Reply:
It would probably be better if we didn't need the Context, but I think it's fine since the ApplicationContext is enough.

https://developer.android.com/topic/libraries/architecture/viewmodel#implement
If the ViewModel needs the Application context, for example to find a system service, it can extend the AndroidViewModel class and have a constructor that receives the Application in the constructor, since Application class extends Context.

Reply:
I think that's better than adding complexity by going through the Activity just to load the string and going back to the ViewModel.